pkgs/applications/misc/survex/default.nix

```diff
@@ -14,34 +15,52 @@
 , xlibsWrapper
 , docbook2x
 , docbook5
+, Carbon
+, Cocoa
 }:
 
+let
+  perlenv = perl532.withPackages (perlPackages: with perlPackages; [ LocalePO ] );
+in
 stdenv.mkDerivation rec {
   pname = "survex";
   version = "1.2.44";
 
-  nativeBuildInputs = [ docbook5 x11 libGL libGLU docbook2x autoreconfHook pkg-config wxGTK30-gtk3 ffmpeg proj python (perl532.withPackages (perlPackages: with perlPackages; [ LocalePO ] )) ];
+  nativeBuildInputs = [ docbook5 docbook2x autoreconfHook pkg-config perlenv python ];
+
+  buildInputs = [
+    libGL libGLU ffmpeg proj
+  ] ++ lib.optionals stdenv.hostPlatform.isDarwin [
+    wxmac Carbon Cocoa
+  ] ++ lib.optionals stdenv.hostPlatform.isLinux [
+    wxGTK30-gtk3 xlibsWrapper
+  ];
 
   src = fetchgit {
     url = "git://git.survex.com/survex";
     rev = version;
     sha256 = "11gaqmabrf3av665jy3mr0m8hg76fmvnd0g3rghzmyh8d8v6xk34";
   };
 
+  enableParallelBuilding = true;
+
   # Docs rely on sgmltools-lite, a package that would be quite complex to
   # provide as it is quite old. So this preConfigure hook effectively disables
   # the doc generation. An example of packaging sgmltools-lite from Gentoo can
   # be found here:
   # https://gitweb.gentoo.org/repo/gentoo.git/tree/app-text/sgmltools-lite/sgmltools-lite-3.0.3-r15.ebuild?id=0b8b716331049599ea3299981e3a9ea6e258c5e0
 
-  preConfigure = ''
+  postPatch = ''
+    patchShebangs .
     echo "" > doc/Makefile.am
-  '';
-
-  postConfigure = ''
     # substituteInPlace doc/Makefile --replace "docbook2man" "docbook2man --sgml" # Will be needed once sgmltools-lite is packaged.
+    for perltool in './extract-msgs.pl' './gettexttomsg.pl' '$(srcdir)/gdtconvert' '$(srcdir)/gen_img2aven'; do
+      substituteInPlace src/Makefile.am \
+        --replace "$perltool" "${perlenv}/bin/perl $perltool"
+    done
+    substituteInPlace lib/Makefile.am \
+      --replace '$(srcdir)/make-pixel-font' '${perlenv}/bin/perl $(srcdir)/make-pixel-font'
```

Comment:
@OPNA2608 What exactly does this do differently? My solution seems to work, but this is much more complex. What is the benefit from the added complexity?

Reply:
Oh, I wanted to leave a link to the issue but I forgot about it. Same problem as

> `# Fixes shebang -> wrapper problem on Darwin`

in unscii, relevant issue is #65351 (seems to apply to `perl.withPackages` too).

Those perl scripts don't execute properly on Darwin. The platform doesn't seem to like when a shebang for an executable file points to a shell script (like the wrapper that `perl.withPackages` produces so its perl binary can see the packages), so it tries to run those files with the regular shell instead. It's therefore necessary to manually call them with the perl interpreter so their shebangs don't get tried & ignored.

Reply:
If I remove those 6 lines again:

![Screenshot_macOS_2021-06-09_19:14:27](https://user-images.githubusercontent.com/23431373/121399708-fd5c7f00-c956-11eb-9e49-1e80ad79e44d.png)